sdks/python/apache_beam/io/gcp/dicomio.py

```diff
@@ -167,33 +167,40 @@ class DicomSearch(PTransform):
     }
 
   """
-  def __init__(self, buffer_size=8, max_workers=5, credential=None):
+  def __init__(
+      self, buffer_size=8, max_workers=5, client=None, credential=None):
     """Initializes DicomSearch.
     Args:
       credential: # type: Google credential object, if it is specified, the
       Http client will use it to create sessions instead of the default.
     """
     self.buffer_size = buffer_size
     self.max_workers = max_workers
+    if not client:
+      self.client = DicomApiHttpClient()
+    else:
+      self.client = client
```

Comment:
it's not that important, but you can do `self.client = client or DicomApiHttpClient()`

Feel free to add it if you have the time.

Reply:
Added